actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -257,7 +312,7 @@ def to_h
     #   params.to_unsafe_h
     #   # => {"name"=>"Senjougahara Hitagi", "oddity" => "Heavy stone crab"}
     def to_unsafe_h
-      convert_parameters_to_hashes(@parameters, :to_unsafe_h)
+      permit!.to_h
```

Comment:
I don't think this should mutate the original params object.. otherwise `params.to_unsafe_h` in one place that's being careful some other way could leave a later `params.to_h` exposed to danger.